src/function.ts

```diff
@@ -64,13 +116,153 @@ export interface IFunction<P, O>
     ConditionalFunction<P, O>
   >;
 
+  /**
+   * Event Source for the {@link Function} onSuccess async invocation destination.
+   *
+   * For Lambda, the onSuccess destination is not enabled by default.
+   * It must first be configured via either the {@link Function} constructor
+   * or by using {@link IFunction.configureAsyncInvoke} and that destination must match the bus provided here.
+   *
+   * ```ts
+   * const bus = new EventBus(stack, 'bus');
+   * new Function(stack, 'func', { onSuccess: bus }, async () => {});
+   * ```
+   *
+   * or
+   *
+   * ```ts
+   * const bus = new EventBus(stack, 'bus');
+   * const func = new Function(stack, 'func', async () => {});
+   * // if onSuccess or onFailure is already set, this will fail.
+   * func.configureAsyncInvoke({ onSuccess: bus });
```

Comment:
Hmm configureAsyncInvoke isn't telling me exactly what it means. This is enabling async invoke and routing events to the bus? Maybe `enableAsyncInvoke`?

Reply:
This is the name from the CDK, which I agree, is also confusing..will take another pass.